### PR TITLE
model: fast build fields to dockerInfo.buildDetails [ch1136]

### PR DESCRIPTION
**Pull request by maiamcc**
Hello @jazzdan, @nicks,

Please review the following commits I made in branch maiamcc/fast-build:

8d94a3f1414dcdf67fe240ef07d67b96cf270ff5 (2018-12-21 16:10:23 -0500)
model: fast build fields to dockerInfo.buildDetails

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

File: internal/model/docker_info.go [hunk 1]
```diff
@@ -50,3 +50,17 @@ type FastBuild struct {
 
 func (FastBuild) buildDetails()  {}
 func (fb FastBuild) Empty() bool { return reflect.DeepEqual(fb, FastBuild{}) }
+
+// func (m1 Manifest) stepsEqual(s2 []Step) bool {
```
Comment (jazzdan) [inline internal/model/docker_info.go]: 👀

File: internal/model/manifest.go [hunk 2]
```diff
@@ -104,24 +97,33 @@ func (m Manifest) Dockerignores() []Dockerignore {
 func (m Manifest) LocalPaths() []string {
 	if sbInfo := m.StaticBuildInfo(); !sbInfo.Empty() {
```
Comment (jazzdan) [inline internal/model/manifest.go]: Could we transform this in to a case statement?

File: internal/model/manifest.go [hunk 3]
```diff
@@ -104,24 +97,33 @@ func (m Manifest) Dockerignores() []Dockerignore {
...
 func (m Manifest) Validate() error {
 	if m.Name == "" {
 		return fmt.Errorf("[validate] manifest missing name: %+v", m)
 	}
-	for _, m := range m.Mounts {
-		if !filepath.IsAbs(m.LocalPath) {
-			return fmt.Errorf(
-				"[validate] mount.LocalPath must be an absolute path (got: %s)", m.LocalPath)
+
+	if fbInfo := m.FastBuildInfo(); !fbInfo.Empty() {
```
Comment (jazzdan) [inline internal/model/manifest.go]: This is a lot of indentation, could we move the check about?

```go
if fbInfo := m.FastBuildInfo(); fbInfo.Empty() {
    return nil
}
// ... 
```

File: internal/model/manifest.go (hunk 2 above)
Comment (maiamcc) [reply internal/model/manifest.go]: not really, cuz we're looking at the type of two different fields :-/

**Pull request merged**